src/Workspaces/Core/Portable/Options/OptionServiceFactory.cs

```diff
@@ -202,9 +204,9 @@ public override object GetOption(OptionKey optionKey)
                     return _underlyingOptions.GetOption(optionKey);
                 }
 
-                public override OptionSet WithChangedOption(OptionKey optionAndLanguage, object value)
+                public override OptionSet WithChangedOption(OptionKey optionAndLanguage, object? value)
                 {
-                    return new DocumentSpecificOptionSet(_documentOptions, _underlyingOptions, _values.Add(optionAndLanguage, value));
+                    return new DocumentSpecificOptionSet(_documentOptions, _underlyingOptions, _values.SetItem(optionAndLanguage, value));
```

Comment:
I'm surprised we haven't hit any bugs because this was `Add` instead of `SetItem`...

Reply:
So am I!